packages/engine/Specs/Scene/PrimitiveCollectionSpec.js

```diff
@@ -649,6 +649,51 @@ describe(
         primitives.lowerToBottom(p);
       }).toThrowDeveloperError();
     });
+
+    it("fires an event on primitive added", function () {
+      const p = createLabels();
+      const primitiveAdded = primitives.primitiveAdded;
+
+      const onPrimitiveAddedSpy = jasmine.createSpy("event");
+      primitiveAdded.addEventListener(onPrimitiveAddedSpy);
+      expect(onPrimitiveAddedSpy).not.toHaveBeenCalled();
+      primitives.add(p);
+      expect(onPrimitiveAddedSpy).toHaveBeenCalledOnceWith(p);
+    });
+
+    it("fires an event on primitive removed", function () {
+      const p = createLabels();
+      const primitiveRemoved = primitives.primitiveRemoved;
+
+      const onPrimitiveRemovedSpy = jasmine.createSpy("event");
+      primitiveRemoved.addEventListener(onPrimitiveRemovedSpy);
+      expect(onPrimitiveRemovedSpy).not.toHaveBeenCalled();
+      primitives.add(p);
+      expect(onPrimitiveRemovedSpy).not.toHaveBeenCalled();
+      primitives.remove(p);
+      expect(onPrimitiveRemovedSpy).toHaveBeenCalledOnceWith(p);
+    });
+
+    it("fires events on primitive remove all", function () {
+      const collection = new PrimitiveCollection({ destroyPrimitives: false });
```

Comment:
I'm manually destroying as the `toHaveBeenCalledWith()` test was triggering a 'this primitive has been destroyed' error.

Reply:
Should this be noted in the `primitiveRemoved` docs? This seems something likely for devs to run into during development as well.